tests/unit/test_platforms_get_platform.py

```diff
@@ -0,0 +1,199 @@
+    # A platform name is present, and no clashing cylc7 configs are:
+    mock_glbl_cfg(
+        'cylc.flow.platforms.glbl_cfg',
+        '''
+        [platforms]
+            [[mace]]
+                hosts = mace001, mace002
+                batch system = slurm
+        '''
+    )
+    task_conf = {'platform': 'mace'}
+    platform = get_platform(task_conf)
+    assert platform['hosts'] == ['mace001', 'mace002']
+    assert platform['batch system'] == 'slurm'
+
+
+@pytest.mark.parametrize(
+    'task_conf, expected_platform_name',
+    [
+        (
+            {
+                'remote': {'host': 'cumin'},
+                'job': {'batch system': 'slurm'}
+            },
+            'ras_el_hanout'
+        ),
+        (
+            {'remote': {'host': 'cumin'}},
+            'spice_bg'
+        ),
+        (
+            {'job': {'batch system': 'batchyMcBatchFace'}},
+            'local_batch_system'
+        ),
+        (
+            {'script': 'true'},
+            'localhost'
+        ),
+        (
+            {
+                'remote': {'host': 'localhost'},
+                'job': {
+                    'batch system': None,
+                    'batch submit command template': None,
+                    'execution polling intervals': None
+                }
+            },
+            'localhost'
+        ),
+        (
+            {
+                'remote': {'host': 'cylcdevbox'},
+                'job': {
+                    'batch system': None,
+                    'batch submit command template': None,
+                    'execution polling intervals': None
+                }
+            },
+            'cylcdevbox'
+        )
+    ]
+)
+def test_get_platform_using_platform_from_job_info(
+    mock_glbl_cfg, task_conf, expected_platform_name
+):
+    """Calculate platform from Cylc 7 config: n.b. If this fails we don't
+    warn because this might lead to many thousands of warnings
+
+    This should not contain a comprehensive set of use-cases - these should
+    be coverend by the unit tests for `platform_from_host_items`
+    """
+    mock_glbl_cfg(
+        'cylc.flow.platforms.glbl_cfg',
+        '''
+        [platforms]
+            [[ras_el_hanout]]
+                hosts = rose, chilli, cumin, paprika
+                batch system = slurm
+            [[spice_bg]]
+                hosts = rose, chilli, cumin, paprika
+            [[local_batch_system]]
+                hosts = localhost
+                batch system = batchyMcBatchFace
+            [[cylcdevbox]]
+                hosts = cylcdevbox
+        '''
+    )
+    assert get_platform(task_conf)['name'] == expected_platform_name
+
+
+def test_get_platform_warn_mode(caplog):
+    task_conf = {
+        'remote': {'host': 'cylcdevbox'},
+        'job': {
+            'batch system': 'pbs',
+            'batch submit command template': 'some template'
+        }
+    }
+    output = get_platform(task_conf, warn_only=True)
+    for forbiddenitem in (
```

Comment:
I know this is not from this pr really, since you are just moving this test out, but I find that a forbiddenitem a bit hard to read.
```suggestion
    for forbidden_item in (
```